test/clj/game/core/runs_test.clj

```diff
@@ -440,3 +440,154 @@
+        (is (= "You accessed Advanced Assembly Lines." (-> (get-runner) :prompt first :msg)) "Accessed A again")
+        (click-prompt state :runner "No action")
+        (is (= "You accessed Chiyashi." (-> (get-runner) :prompt first :msg)) "Accessed C")
+        (click-prompt state :runner "No action")
+        (is (= "You accessed DNA Tracker." (-> (get-runner) :prompt first :msg)) "Accessed D")
+        (click-prompt state :runner "No action")
+        (is (empty? (:prompt (get-runner))) "No more accesses")
+        (is (= "Advanced Assembly Lines" (-> (get-corp) :deck first :title)) "A on top")))))
+
+(deftest-pending multi-access-shuffle
+  ; Currently, shuffling during the accesses is only handled for Bacterial Programming using
+  ; a special flag. This should be generalized at some point.
+  (testing "Correct handling of multi accesses with shuffle in between accesses"
+    (testing "Shuffle from paid ability during accesses"
+      (do-game
+        (new-game {:corp {:hand ["Advanced Assembly Lines" "Brainstorm" "Chrysalis" "DNA Tracker" "Efficiency Committee"
+                                 "Shannon Claire"]
+                          :deck []}})
+        (core/move state :corp (find-card "Advanced Assembly Lines" (:hand (get-corp))) :deck)
+        (core/move state :corp (find-card "Brainstorm" (:hand (get-corp))) :deck)
+        (core/move state :corp (find-card "Chrysalis" (:hand (get-corp))) :deck)
+        (core/move state :corp (find-card "DNA Tracker" (:hand (get-corp))) :deck)
+        (core/move state :corp (find-card "Efficiency Committee" (:hand (get-corp))) :deck)
+        (play-from-hand state :corp "Shannon Claire" "New remote")
+        (let [sc (get-content state :remote1 0)]
+          (core/rez state :corp sc)
+          ; Deck is now ABCD from top to bottom
+          (take-credits state :corp)
+          (run-on state :rd)
+          (core/access-bonus state :runner :rd 3)
+          (run-continue state :access-server)
+          (is (= "You accessed Advanced Assembly Lines." (-> (get-runner) :prompt first :msg)) "Accessed A")
+          (click-prompt state :runner "No action")
+          (is (= "You accessed Brainstorm." (-> (get-runner) :prompt first :msg)) "Accessed B")
+          (click-prompt state :runner "No action")
+          (is (= "You are encountering Chrysalis. Allow its subroutine to fire?" (-> (get-runner) :prompt first :msg)) "Accessing C")
+          (card-ability state :corp sc 1)
+          (click-prompt state :corp "Efficiency Committee")
+          ; R&D was shuffled
+          (click-prompt state :runner "No") ; not letting sub fire
+          (click-prompt state :runner "No action") ; end accessing C
+          (is (= (str "You accessed " (-> (get-corp) :deck first :title) ".")
+                 (-> (get-runner) :prompt first :msg))
+              "Accessing top card of R&D")
+          (click-prompt state :runner "No action")
+          (is false "no")
```

Comment:
👀

Reply:
Whoops. I was using that line to make the test fail so I can see the output even if it would succeed :joy: